speech.js

```diff
@@ -134,41 +192,194 @@ class Speech {
+        return response.json();
+      })
+      .catch((error) => {
+        throw new Error(`[fetchVoices] ${error.message}`);
+      });
+  }
+
+  async fetchVoice(voice) {
+    const url = `${_BASE_URL}${_VOICE_ENDPOINT.replace('{id}', voice)}`;
+    return fetch(url, {
       headers: this._getHeaders(),
-      method: 'GET'
-    }).then(response => response.json());
+      method: "GET",
+    })
+      .then(async (response) => {
+        await this._handle_response_errors(response);
+        return response.json();
+      })
+      .catch((error) => {
+        throw new Error(`[fetchVoice] ${error.message}`);
+      });
   }
 
+  async createVoice(name, enhance, filenames, type = 'instant', gender = null, description = null) {
+    if (name.length === 0) {
+      throw new Error('[Speech.createVoice] Name must be non-empty.');
+    }
+    if (filenames.length === 0) {
+      throw new Error('[Speech.createVoice] Filenames must be non-empty.');
+    }
+
+    const metadata = JSON.stringify({
+      name,
+      enhance,
+      type,
+      gender,
+      description,
+    });
+
+    const formData = new FormData();
+    formData.append('metadata', metadata, {
+        headers: {
+          'Content-Type': 'application/json', 'Content-Disposition': 'form-data; name="metadata"'
+        }
+    });
+    filenames.forEach((filename) => {
+        formData.append('file_field', fs.createReadStream(filename), {
+            filename: path.basename(filename),
+            headers: {
+                'Content-Type': 'application/octet-stream', 'Content-Disposition': 'form-data; name="file_field"'
+            }
+        });
+    });
+
+    return axios.post(`${_BASE_URL}${_CREATE_VOICE_ENDPOINT}`, formData, {
+        headers: {
+            ...formData.getHeaders(),
+            ...this._getHeaders(),
+        },
+    })
+    .then((response) => {
+      if (response.status !== 200) {
+        throw new SpeechError(response.status, response.data);
+      }
+      return response.data;
+    })
+    .catch((error) => {
+      throw new Error(`[createVoice] ${error.message}`);
+    });
+  }
+
+  async updateVoice(voice, options = {}) {
+    const requestOptions = {
+      name: options.name || undefined,
+      starred: options.starred !== undefined ? options.starred : undefined,
+      gender: options.gender || undefined,
+      description: options.description || undefined,
+    };
+    const url = `${_BASE_URL}${_VOICE_ENDPOINT.replace('{id}', voice)}`;
+    return fetch(url, {
+      method: 'PUT',
+      headers: this._getHeaders(),
+      body: JSON.stringify(requestOptions),
+    })
+    .then(async (response) => {
+      await this._handle_response_errors(response);
+      return response.json();
+    })
+    .catch(error => {
+      throw new Error(`[updateVoice] ${error.message}`);
+    });
+  }
+
+  async deleteVoice(voice) {
+    const url = `${_BASE_URL}${_VOICE_ENDPOINT.replace('{id}', voice)}`;
```

Comment:
The general pattern here (fetch, get response w/ error handling and return json, catch errors) appears in many of the methods in this file times and could maybe be factored to an internal helper and used here and elsewhere, with varying endpoint/params/message-prefix params. Just a thought, and could do in subsequent PR.